plugins/french/xiaowaz.ts

```diff
@@ -10,52 +10,74 @@ class XiaowazPlugin implements Plugin.PluginBase {
   name = 'Xiaowaz';
   icon = 'src/fr/xiaowaz/icon.png';
   site = 'https://xiaowaz.fr';
-  version = '1.0.0';
+  version = '1.0.1';
   filters: Filters | undefined = undefined;
 
   async getCheerio(url: string): Promise<CheerioAPI> {
-    const r = await fetchApi(url, {
-      headers: { 'Accept-Encoding': 'deflate' },
-    });
+    const r = await fetchApi(url);
     const body = await r.text();
     const $ = load(body);
     return $;
   }
 
-  async popularNovels(
-    pageNo: number,
-    {
-      showLatestNovels,
-      filters,
-    }: Plugin.PopularNovelsOptions<typeof this.filters>,
-  ): Promise<Plugin.NovelItem[]> {
-    if (pageNo > 1) return [];
-
-    const novels: Plugin.NovelItem[] = [];
+  async getAllNovels(): Promise<Plugin.NovelItem[]> {
+    let novels: Plugin.NovelItem[] = [];
     let novel: Plugin.NovelItem;
-    let url = this.site;
-    let $ = await this.getCheerio(url);
-    $('li.page_item.page-item-866.page_item_has_children ul.children li').each(
-      (i, elem) => {
-        let novelName = $(elem).first().text().trim();
-        if (novelName.charAt(novelName.length - 1) === '✔') {
-          novelName = novelName.slice(0, -1);
-        }
+    let $ = await this.getCheerio(this.site);
+
+    const categories: Cheerio<Element>[] = [
+      $('li.page_item').find('a:contains("Séries")').parent(),
+      $('li.page_item').find('a:contains("Créations")').parent(),
+      $('li.page_item').find('a:contains("†")').parent(),
+    ];
+
+    categories.forEach(cheerio => {
+      cheerio.find('ul.children li').each((i, elem) => {
+        const novelName = $(elem).first().text().trim().replace('✔', '');
         const novelUrl = $(elem).find('a').attr('href');
 
-        if (novelUrl && novelName) {
+        // Douluo Dalu is no good
+        if (novelUrl && novelName && novelName !== 'Douluo Dalu') {
           novel = {
             name: novelName,
-            cover: defaultCover,
             path: novelUrl.replace(this.site, ''),
           };
           novels.push(novel);
         }
-      },
-    );
+      });
+    });
+    return novels;
+  }
+
+  async getNovelsCovers(
+    novels: Plugin.NovelItem[],
+  ): Promise<Plugin.NovelItem[]> {
+    for (const novel of novels) {
+      const $novel = await this.getCheerio(this.site + novel.path);
+      novel.cover =
+        $novel('img[fetchpriority = "high"]').first().attr('src') ||
+        $novel('img.aligncenter').first().attr('src') ||
+        defaultCover;
+    }
     return novels;
   }
 
+  async popularNovels(pageNo: number, {}: any): Promise<Plugin.NovelItem[]> {
+    const maxPages = 4;
```

Comment:
I think you should use pageSize instead. 
example:
```js
const pageSize = 10;
const maxPage = Math.ceil(novels.length/pageSize);
if (pageNo  > maxPage) return [];
```

Reply:
I wanted to return before fetching all the novels, but I guess I can do the same with a static, I will make even less requests